lib/tasks/vmodl_helper.rb

```diff
@@ -86,35 +86,24 @@ def verify!
 
   def generate!
     wsdl_types_by_name.each_value do |type|
-      type_name  = type.name.name
-      vmodl_data = @vmodl[type_name]
+      type_name = type.name.name
 
-      if vmodl_data.nil?
-        base_class           = wsdl_types_by_name[type.complexcontent.extension.base.name]
-        inherited_properties = base_class.elements.map { |element| element.name.name }
-        properties           = type.elements.reject { |e| inherited_properties.include?(e.name.name) }
-
-        vmodl_data = {
-          'kind'      => 'data',
-          'props'     => properties.map do |element|
-            {
-              'name'           => element.name.name,
-              'is-optional'    => element.minoccurs == 0,
-              'is-array'       => element.maxoccurs != 1,
-              'version-id-ref' => nil,
-              'wsdl_type'      => wsdl_to_vmodl_type(element.type)
-            }
-          end,
-          'wsdl_base' => type.complexcontent.extension.base.name
-        }
-
-        @vmodl[type_name] = vmodl_data
-        @vmodl['_typenames']['_typenames'] << type_name
-
-        puts "Adding #{type_name} to vmodl"
-
-        wsdl_to_rbvmomi_namespace(type).loader.add_types type_name => vmodl_data
+      # Update the existing vmodl object if it exists otherwise instantiate a
+      # new one.
+      vmodl_data = @vmodl[type_name] || build_wsdl_obj!(type)
+      props_by_name = vmodl_data['props'].index_by { |prop| prop['name'] }
```

Comment:
Is it possible to have multiple props with the same name?  I'm guessing no, but that case wouldn't be handled here.

Reply:
No it isn't